src/libraries/System.ComponentModel.TypeConverter/src/System/ComponentModel/TypeDescriptor.cs

```diff
@@ -816,12 +825,19 @@ public static TypeConverter GetConverter(object component, bool noCustomTypeDesc
         /// <summary>
         /// Gets a type converter for the specified type.
         /// </summary>
+        [RequiresUnreferencedCode(TypeConverter.RequiresUnreferencedCodeMessage)]
         public static TypeConverter GetConverter([DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.All)] Type type)
         {
             return GetDescriptor(type, nameof(type)).GetConverter();
         }
 
+        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2026:RequiresUnreferencedCode",
+            Justification = "The callers of this method ensure getting the converter is trim compatible - i.e. the type is not Nullable<T>.")]
+        internal static TypeConverter GetConverterTrimUnsafe([DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.All)] Type type) =>
```

Comment:
>GetConverterTrimUnsafe [](start = 38, length = 22)

Nit: Did you meant to say TrimSafe?

Reply:
No, this is Unsafe. The reason being is because the we are suppressing the warning that `GetConverter` RequiresUnreferencedCode. So the caller needs to ensure the Type they are passing in is "safe" to get their converter for. Just like when dealing with the Unsafe.As methods, you need to ensure you know what you are doing.

---
In reply to: [596457352](https://github.com/dotnet/runtime/pull/49786#discussion_r596457352) [](ancestors = 596457352)

Reply:
> Wouldn't we want them to get a warning when calling this to make sure that they preserve what needs to be preserved and then suppress the warning after they ensure that what they are doing is safe?

That is exactly what happened. There are 10 places in this assembly that called `TypeDescriptor.GetConverter` directly with a hard-coded Type. All 10 got a warning, and I analyzed that they were passing in primitive types - like `int`, `float`, and `InheritanceLevel` enum. Instead of suppressing 10 different places that said "what I am doing is safe", I made a common method that the 10 places could call into that says "what I am doing is safe".